source/common/config/grpc_mux_impl.cc

```diff
@@ -234,7 +238,7 @@ void GrpcMuxImpl::onDiscoveryResponse(
     error_detail->set_message(Config::Utility::truncateGrpcStatusMessage(e.what()));
   }
   api_state_[type_url].request_.set_response_nonce(message->nonce());
-  queueDiscoveryRequest(type_url);
+  sendDiscoveryRequest(type_url);
```

Comment:
are there any ordering concerns because the queue is bypassed here?

Reply:
No, since `sendDiscoveryResponse` sets `pending_` state and effectively queues when we resume.